Makefile

```diff
@@ -30,15 +30,18 @@ build: ## Builds the all docker setup.
 	docker compose build
 
 run-devserver: ## Runs the dev-server
-	docker compose up angular-build -d
+# Add container label that it hasn't completed first run
+	COMPLETED_FIRST_RUN=0 docker compose up angular-build -d
 	$(MAKE) update.package
 	docker cp oppia-angular-build:/app/oppia/node_modules .
 	docker compose stop angular-build
-	docker compose up dev-server -d --no-deps
+	COMPLETED_FIRST_RUN=0 docker compose up dev-server -d --no-deps
 	$(MAKE) update.requirements
 	$(MAKE) run-offline
 
 run-offline: ## Runs the dev-server in offline mode
+	$(MAKE) check.container-exists
+	$(MAKE) check.dev-server-started-once
 	$(MAKE) start-devserver
 	@echo 'Please visit http://localhost:8181 to access the development server.'
 	@echo 'Check dev-server logs using "make logs.dev-server"'
```

Comment:
Move the echo statements to start-devserver

Reply:
done!